homeassistant/components/lutron_caseta.py

```diff
@@ -0,0 +1,109 @@
+"""
+Component for interacting with a Lutron Caseta system.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/lutron_caseta/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (CONF_HOST,
+                                 CONF_USERNAME,
+                                 CONF_PASSWORD)
+from homeassistant.helpers import discovery
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['https://github.com/gurumitts/'
+                'pylutron-caseta/archive/v0.2.0.zip#'
+                'pylutron-caseta==v0.2.0', 'paramiko==2.1.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+LUTRON_CASETA_SMARTBRIDGE = 'lutron_smartbridge'
+LUTRON_CASETA_DEVICES = 'lutron_devices'
+
+DOMAIN = 'lutron_caseta'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, base_config):
+    """Setup the Lutron component."""
+    from pylutron_caseta.smartbridge import Smartbridge
+
+    hass.data[LUTRON_CASETA_SMARTBRIDGE] = None
+    hass.data[LUTRON_CASETA_DEVICES] = None
+
+    config = base_config.get(DOMAIN)
+
+    hass.data[LUTRON_CASETA_SMARTBRIDGE] = Smartbridge(
+        hostname=config[CONF_HOST],
+        username=config[CONF_USERNAME],
+        password=config[CONF_PASSWORD]
+    )
+    _LOGGER.debug("Connected to Lutron smartbridge at %s",
```

Comment:
Yep.. if it failed to connect what is the best way to handle this situation in HA?  Throw except? A particular HA signal?